bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -67,13 +93,32 @@ public void propertyChange(PropertyChangeEvent evt) {
 				mm.addMarker(false, this, Severity.Warning, String.format("Tracking: target reaches %1.2f\u00B0.", maxElevation), v, a);
 			}
 
+			// Function to create markers that report solver intervals.
+			final BiFunction<String, String, Option<Marker>> createSolverMarker =
+					(cacheName, prefix) -> {
+						final Map<Obs, Union<Interval>> c = v.getSchedule().getCache(cacheName);
+						final Union<Interval>          ws = c.getOrDefault(a.getObs(), new Union<>());
+						final ImList<Interval>         is = DefaultImList.create(ws.getIntervals());
+
+						if (is.isEmpty()) {
+							return None.instance();
+						} else {
+							final String m = is.map(i -> formatInterval(zone, i)).mkString(prefix + " ", ", ", ".");
+							return new Some<>(new Marker(false, this, Severity.Notice, m, v, a));
+						}
+					};
+
 			// Timing window
 			int percentVisible = (int) (100 * a.getMean(Circumstance.TIMING_WINDOW_OPEN, false));
 			switch (percentVisible) {
 			case 100: // good!
-				if (a.getObs().getTimingWindows().size() > 0)
+				if (a.getObs().getTimingWindows().size() > 0) {
 					mm.addMarker(true, this, Severity.Info, "Timing constraint is met.", v, a);
-				break;
+
+					// Report timing windows
+					createSolverMarker.apply("timingUnionCache", "Must be in the timing window").foreach(m -> mm.addMarker(m));
```

Comment:
We should define these cache names like `"timingUnionCache"` somewhere as we discussed.  Maybe in `Variant` since that's where they are created?

Reply:
Yes, these should be constants on `Variant`.